src/csync/csync_exclude.cpp

```diff
@@ -320,13 +320,24 @@ bool ExcludedFiles::reloadExcludeFiles()
     bool success = true;
     const auto keys = _excludeFiles.keys();
     for (const auto& basePath : keys) {
-        for (const auto &excludeFile : _excludeFiles.value(basePath)) {
+        auto itValue = _excludeFiles.find(basePath);
+        if (itValue == std::end(_excludeFiles)) {
+            continue;
+        }
+        auto &excludeFiles = *itValue;
+        for (auto excludeFileIt = std::begin(excludeFiles); excludeFileIt != std::end(excludeFiles); ) {
+            const auto &excludeFile = *excludeFileIt;
             QFile file(excludeFile);
-            if (file.exists() && file.open(QIODevice::ReadOnly)) {
-                loadExcludeFilePatterns(basePath, file);
+            if (!file.exists()) {
+                excludeFileIt = excludeFiles.erase(excludeFileIt);
```

Comment:
@matthieugallien All this seems complicated. Can't you just loop through `_excludeFiles` and remove non-existing exclude files before looping them for `loadExcludeFilePatterns(basePath, file);`? This will double the loop, but, will improve readability.